### ISSUE TITLE
all: define checker stabilization steps

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#424]
# Stabilization

## Overview

Experimental checker is one that has `attrExperimental` attribute. It means that checker is disabled by default. To remove such attribute and make checker enabled by default, stabilization process must take place (described below).

## Stabilization process and requirements

For every checker to be stabilized, an issue should be created:
```
lint: make ${checkerName} checker stable
```

All requirements below are mandatory:

1. Checker name must be finalized. It's far less pleasant to rename stable checker.
1. Checker documentation must be well-written, easy-to-understand and examples must "sell" the checker in the best way they could.
1. Warnings generated are clear while succinct (and grammatically correct).
1. Both negative and positive test suite is does cover checker well. You can verify that by using coverage report. There is no constraint like "no less than 90% of coverage", but keep in mind that good coverage helps when we're fixing bugs or make checker capable of finding more issues.
1. The checker must be tested on at least 3 big projects. One of them can be `golang/go` (GOROOT). The aim is to see whether there are non-obvious false-positives on a big scale, in "real world code".
1. The "opinionated" or not decision must be made. If there are doubts, do a survey or other kind of investigation that can reveal yes-vs-no ratio. It's not recommended to move checker to stable group and say that "opinionated" tag is temporary. You just don't move that checker to a stable group until you pretty sure that this one is opinionated or not.
1. Checker is capable of detecting most issues it should detect. It should be useful for the tasks it was made to solve. Experimental checkers have advantage that we can remove and/or replace them, merge functionality and so on. If checker is stable, we loose that freedom.
1. Checker is tuned to have few false-positives. If the checker can't have low false-positives rate by design, it can't be marked as non-experimental. Such checkers should be disabled by default.

**Comment by cristaloleg** [comment]
List of checkers to be easily(?) stabilized:
- appendAssign
- deadCodeAfterLogFatal
- defaultCaseOrder
- deferInLoop
- dupCase
- emptyFmt
- importShadow
- regexpMust
- sloppyLen
- stdExpr
- unusedParam

**Comment by Quasilyte** [comment]
> deadCodeAfterLogFatal

This name is temporary and will be changed when it will handle not only `Log.Fatal`.
Checkers with temporary names can't be stabilized.

**Comment by cristaloleg** [comment]
I've mentioned 'looks good' (for me) checkers, name are just names 😉

**Comment by Quasilyte** [comment]
Well, names **are** one of stabilization requirements.
We shouldn't rename checkers after they are stabilized.
This is the whole point.

**Comment by cristaloleg** [comment]
Hm...I was thinking from checkers behaviour and how they work.

**Comment by Quasilyte** [comment]
I've updated first message.
Can anyone help me to figure out the missing pieces?

**Comment by cristaloleg** [comment]
LGTM

**Comment by Quasilyte** [comment]
+1 requirement: good docs.
